src/utils.h

```diff
@@ -205,7 +205,10 @@ static inline int get_index_hash(void *p, int max, int struct_size, uint32_t key
 
 #define FAIL(a, ...)               \
 	{                              \
-		LOGL(0, a, ##__VA_ARGS__); \
+		if (opts.log) {            \
```

Comment:
If oyu do not enable logging, you will not get even fatal exception which is not right IMO.

Reply:
> If oyu do not enable logging, you will not get even fatal exception which is not right IMO.

Just the opposite, again!
Check the _full_ code:
```
		if (opts.log) {            \
			LOGL(0, a, ##__VA_ARGS__); \
		} else                     \
			LOG0(a, ##__VA_ARGS__);    \
```
When no log is enabled the log is unconditional (`LOG0`) and when it's enabled then it's equal to the general log (`LOGL`).
So the code is correct.

Please, compile with the patch and test it! It works with/without logs.
Regards.

Reply:
Please @catalinii comment about it!